# ANNOTEX
ANNOTEX - Annotation Extension for ChimeraX

https://www.biorxiv.org/content/10.1101/2023.08.27.554996v2

# Installation and Use of  ANNOTEX
After successful installation of ANNOTEX, using command “devel build PATH_TO_ANNOTATER_FOLDER; devel install PATH_TO_ANNOTATER_FOLDER; devel clean PATH_TO_ANNOTATER_FOLDER” in the ChimeraX command line, the user can load the previously created proteome predictions, by selecting the “Data folder” path and pressing the “Search” button (Supplementary Figure 3b).

The structure prediction for the first protein-coding gene will be displayed in the ChimeraX viewer, and the tool window will show the gene ID, a potential eggNOG annotation with a description of the protein function and E-value, if applicable, and a DeepTMHMM prediction that can be a global protein (GLOB), a TMD, a SP, or both TMD and SP. Below these descriptions, the structural matches from PDB (blue), AlphaFold microsporidian proteomes, and the available 20 model organisms (red) and AlphaFold Swissprot (yellow) are listed followed by the Diamond sequence-based blast hits (Supplementary Figure 3b). All entries are sorted according to their assigned E-value which can help to assess the certainty of a structural or sequence-based match.

The protein of interest is displayed as a complete model #1 in rainbow colors according to the AlphaFold confidence (Supplementary Figure 3a, d left panel). If the domains’ position relative to each other is of high uncertainty, the model can be hidden in the Models window (Supplementary Figure 3a, lower panel) and the individual, well-predicted domains become visible. Further, single domains in the list can have a subcategory that indicates a “missing structure” due to the exclusion/removal of highly disordered regions or flexible linkers. To visually inspect the structural matches, the user can click on an entry and subsequently, the corresponding protein structure will be superimposed (automatically applying the matchmaker in ChimeraX) to the investigated protein in the model viewer (Supplementary Figure 3a, d right panel). The superimposed structure can either be a single protein or a protein complex. In the latter case, the protein of the complex with structural similarity to the protein of interest will be highlighted in green.

If the structural match is from a PDB entry, the protein “chain” of the complex matching the protein of interest is added as a suffix to the PDB ID as shown in Supplementary Figure 3a (PDB 4pz6_A) and d (PDB 3j96_G). Further, for any selected structural hit an info panel will appear in the console log listing “ID”, “Title”, “Chain information” and “Parameters” (Supplementary Figure 3c). Once the user has found an adequate functional annotation, enabling the checkbox “Overwrite annotation on click” allows the protein being investigated to be named after the matching protein by clicking on the corresponding entry (Supplementary Figure 3b). The complete genome annotation can be saved in tsv-format facilitating accessibility of the file and further editing in a suitable program. ANNOTEX was successfully tested using ChimeraX version 1.3 and 1.4.

![Figure_S3_Annotatex_V5_23 11 2023](https://github.com/Barandun-Lab/ANNOTEX/assets/22501102/4f7778dc-8322-4417-8783-0408ea1f3726)

# Generating a database for ANNOTEX
For the functional annotation, a database was generated to retrieve the best sequence and structure-based matches for each input sequence. The sequence-based search was done using Diamond with the ultra-sensitive option against the non-redundant NCBI database. The eggNOG mapper was used with non-default parameters (Percentage identity: 15%, Minimum hit bit-score: 40) and allowed for functional annotation based on orthology predictions which is considered more precise than traditional homology searches. For structural matches we folded the V. necatrix proteome and the hypothetical proteins of E. cuniculi using ColabFold with default parameters.

Next, we used each individual predicted 3D structure as input for Foldseek searches employing the alignment type 3Di+AA Gotoh-Smith-Waterman (local, default) and ran it against three different databases: 1. PDB, 2. AlphaFold database from the 20 first annotated model organisms (accession date: 07-15-2022), one representative of each microsporidian clade (Figure 1a), and 3. SwissProt AlphaFold. Additionally, for the E. cuniculi proteins, individual, well-predicted protein domains were automatically separated using the Predicted Aligned Error (PAE) (45) and subjected to the TM-align algorithm in Foldseek. As a measure of confidence, the E-value is displayed for all Diamond and eggNOG searches, while the significance of the Foldseek searches varies with the alignment type: The bit score assesses 3Di+AA Gotoh-Smith-Waterman search results and the TM score (global score) represents the confidence of TM-align searches. Further, to predict the overall 3D structure and the presence of a SP or TMD for each analyzed protein, the Deep Transmembrane Helix Hidden Markov Model (DeepTMHMM) (1.0.20) software (93) was used. 

To combine and display the generated information and similarity matches for each V. necatrix input sequence, we developed a ChimeraX annotator plugin, that we named ANNOTEX (Supplementary Figure 3). It retrieves a list of all predicted V. necatrix protein 3D structures, shows the eggNOG annotation in the user interface (Supplementary Figure 3b), and presents a list of structural matches and sequence-based hits, respectively, along with corresponding confidence values. Further, the proteins corresponding to structural hits can be superimposed with the V. necatrix protein of interest, allowing for visual inspection of the structure match. Additionally, the overview of all structural and sequence hits per protein allows for manual curation and functional annotation according to the best match.
